contrib/hyperscan/matching/input_matchers/source/matcher.cc

```diff
@@ -1,91 +1,55 @@
 #include "contrib/hyperscan/matching/input_matchers/source/matcher.h"
 
-#include <numeric>
-
 namespace Envoy {
 namespace Extensions {
 namespace Matching {
 namespace InputMatchers {
 namespace Hyperscan {
 
-ScratchThreadLocal::ScratchThreadLocal(const hs_database_t* database) {
+ScratchThreadLocal::ScratchThreadLocal(const hs_database_t* database,
+                                       const hs_database_t* som_database) {
   hs_error_t err = hs_alloc_scratch(database, &scratch_);
   if (err != HS_SUCCESS) {
     throw EnvoyException(fmt::format("unable to allocate scratch space, error code {}.", err));
   }
+  if (som_database) {
+    err = hs_alloc_scratch(som_database, &scratch_);
+    if (err != HS_SUCCESS) {
+      throw EnvoyException(
+          fmt::format("unable to allocate start of match scratch space, error code {}.", err));
+    }
+  }
 }
 
-Matcher::Matcher(
-    const envoy::extensions::matching::input_matchers::hyperscan::v3alpha::Hyperscan& config,
-    ThreadLocal::SlotAllocator& tls)
+Matcher::Matcher(const std::vector<const char*>& expressions,
+                 const std::vector<unsigned int>& flags, const std::vector<unsigned int>& ids,
+                 ThreadLocal::SlotAllocator& tls, bool som)
```

Comment:
What's `som`?

Reply:
`som` represents `start of matching`. It is an abbreviation used by Hyperscan. With `som` flag turned on, all the matching will report their start and end position (while increasing matching time), which is used for `CompiledMatcher::replaceAll`.

I can rename to `report_start_of_matching` if it is not clear.

Reply:
Yes that'd be great. See https://google.github.io/styleguide/cppguide.html#General_Naming_Rules